dali/pipeline/pipeline_test.cc

```diff
@@ -311,113 +287,182 @@ TYPED_TEST(PipelineTest, TestSerialization) {
+    auto &output = ws->Output<CPUBackend>(idx);
+    auto tmp_size = output.capacity();
+    output.mutable_data<size_t>();
+    output.Resize({2});
+    auto out = output.mutable_data<size_t>();
+    out[0] = tmp_size;
+    out[1] = input.capacity();
+  }
+};
 
-  // Run the pipeline
-  for (int i = 0; i < 5; ++i) {
-    pipe.RunPrefetch();
-    pipe.RunCopy();
-    pipe.RunForward();
+class DummyPresizeOpGPU : public Operator<GPUBackend> {
+ public:
+  explicit DummyPresizeOpGPU(const OpSpec &spec)
+      : Operator<GPUBackend>(spec) {
+  }
 
-    CUDA_CALL(cudaStreamSynchronize(stream));
+  void RunImpl(Workspace<GPUBackend>* ws, int idx) override {
+    auto &input = ws->Input<GPUBackend>(0);
+    auto &output = ws->Output<GPUBackend>(0);
+    output.mutable_data<size_t>();
+    size_t tmp_size[2] = {output.capacity(), input.capacity()};
+    std::vector< std::vector<Index> > shape {{1}};
+    output.Resize(shape);
+    auto out = output.mutable_data<size_t>();
+    CUDA_CALL(cudaStreamSynchronize(ws->stream()));
+    CUDA_CALL(cudaMemcpy(out, &tmp_size, sizeof(size_t) * 2, cudaMemcpyDefault));
+  }
+};
 
-    // Verify the results
-    this->CompareData(
-        pipe.output_batch().template data<uint8>(),
-        batch->template data<uint8>(),
-        batch->size());
+class DummyPresizeOpMixed : public Operator<MixedBackend> {
+ public:
+  explicit DummyPresizeOpMixed(const OpSpec &spec)
+      : Operator<MixedBackend>(spec) {
   }
-}
 
-TYPED_TEST(PipelineTest, TestNoOps) {
-  int num_thread = TypeParam::nt;
+  using Operator<MixedBackend>::Run;
+  void Run(MixedWorkspace* ws) override {
+    auto &input = ws->Input<CPUBackend>(0, 0);
+    auto &output = ws->Output<GPUBackend>(0);
+    output.mutable_data<size_t>();
+    size_t tmp_size[2] = {output.capacity(), input.capacity()};
+    std::vector< std::vector<Index> > shape {{1}};
+    output.Resize(shape);
+    auto out = output.mutable_data<size_t>();
+    CUDA_CALL(cudaStreamSynchronize(ws->stream()));
+    CUDA_CALL(cudaMemcpy(out, &tmp_size, sizeof(size_t) * 2, cudaMemcpyDefault));
+  }
+};
 
-  int batch_size = this->jpegs_.size();
-  Pipeline pipe(batch_size, num_thread, 0, 0, true);
+DALI_REGISTER_OPERATOR(DummyPresizeOp, DummyPresizeOpCPU, CPU);
+DALI_REGISTER_OPERATOR(DummyPresizeOp, DummyPresizeOpGPU, GPU);
+DALI_REGISTER_OPERATOR(DummyPresizeOp, DummyPresizeOpMixed, Mixed);
+
+DALI_SCHEMA(DummyPresizeOp)
+  .DocStr("Dummy")
+  .NumInput(1)
+  .NumOutput(1);
+
+TEST_F(PipelineTestOnce, TestPresize) {
+  const int batch_size = 1;
+  const int num_thread = 1;
+  const bool pipelined = true;
+  const bool async =  true;
+  DALIImageType img_type = DALI_RGB;
+
+  const int presize_val_CPU = 11;
+  const int presize_val_Mixed = 157;
+  const int presize_val_GPU = 971;
+  const int presize_val_make_contiguous = 55;
+
+  // Create the pipeline
+  Pipeline pipe(
+      batch_size,
+      num_thread,
+      0, -1, pipelined, 3,
+      async,
+      presize_val_make_contiguous);
```

Comment:
Maybe this might be named `presize_val_global` or `presize_val_default` since in this context it is not only applied to `MakeContiguous` op and is checked against `DummyPresizeOpGPU` later?

Reply:
Done